src/fu-engine-integrity.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuEngine"
+
+#include "config.h"
+
+#include "fu-engine-integrity.h"
+
+/* exported for the self tests */
+void
+fu_engine_integrity_add_checksum(GHashTable *self, const gchar *id, const gchar *csum)
+{
+	g_hash_table_insert(self, g_strdup(id), g_strdup(csum));
+}
+
+static void
+fu_engine_integrity_add_measurement(GHashTable *self, const gchar *id, GBytes *blob)
+{
+	g_autofree gchar *csum = g_compute_checksum_for_bytes(G_CHECKSUM_SHA256, blob);
```

Comment:
If we're starting from scratch, any reason to pick SHA256 rather than 384?

Reply:
I don't think it matters; I think MD5 would have been fine too. SHA384 would mean upping our gio dep to 2.51 and SHA1 means I have to explain why I'm using it to Red Hat security.

Reply:
OK